ast/marshal_test.go

```diff
@@ -0,0 +1,1326 @@
+			JSON: `{"location":{"file":"example.rego","row":1,"col":2},"type":"string","value":"example"}`,
+			ExpectedTerm: func() *Term {
+				v, _ := InterfaceToValue("example")
+				return &Term{
+					Value:    v,
+					Location: NewLocation([]byte{}, "example.rego", 1, 2),
+				}
+			}(),
+		},
+	}
+
+	for name, data := range testCases {
+		t.Run(name, func(t *testing.T) {
+			var term Term
+			err := json.Unmarshal([]byte(data.JSON), &term)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !term.Equal(data.ExpectedTerm) {
+				t.Fatalf("expected:\n%#v got\n%#v", data.ExpectedTerm, term)
+			}
+			if data.ExpectedTerm.Location != nil {
+				if !term.Location.Equal(data.ExpectedTerm.Location) {
+					t.Fatalf("expected location:\n%#v got\n%#v", data.ExpectedTerm, term)
+				}
+			}
+		})
+	}
+}
+
+func TestPackage_MarshalJSON(t *testing.T) {
+	testCases := map[string]struct {
+		Package      *Package
+		ExpectedJSON string
+	}{
+		"base case": {
+			Package: &Package{
+				Path: EmptyRef(),
+			},
+			ExpectedJSON: `{"path":[]}`,
+		},
+		"location excluded": {
+			Package: &Package{
+				Path:     EmptyRef(),
+				Location: NewLocation([]byte{}, "example.rego", 1, 2),
+				jsonFields: map[string]bool{
+					"location": false,
+				},
+			},
+			ExpectedJSON: `{"path":[]}`,
+		},
+		"location included": {
+			Package: &Package{
+				Path:     EmptyRef(),
+				Location: NewLocation([]byte{}, "example.rego", 1, 2),
+				jsonFields: map[string]bool{
+					"location": true,
+				},
+			},
+			ExpectedJSON: `{"location":{"file":"example.rego","row":1,"col":2},"path":[]}`,
+		},
+	}
+
+	for name, data := range testCases {
+		t.Run(name, func(t *testing.T) {
+			bs := util.MustMarshalJSON(data.Package)
+			got := string(bs)
+			exp := data.ExpectedJSON
+
+			if got != exp {
+				t.Fatalf("expected:\n%s got\n%s", exp, got)
+			}
+		})
+	}
+}
+
+func TestPackage_UnmarshalJSON(t *testing.T) {
+	testCases := map[string]struct {
+		JSON            string
+		ExpectedPackage *Package
+	}{
+		"base case": {
+			JSON: `{"path":[]}`,
+			ExpectedPackage: &Package{
+				Path: EmptyRef(),
+			},
+		},
+		"location case": {
+			JSON: `{"location":{"file":"example.rego","row":1,"col":2},"path":[]}`,
+			ExpectedPackage: &Package{
+				Path:     EmptyRef(),
+				Location: NewLocation([]byte{}, "example.rego", 1, 2),
+			},
+		},
+	}
+
+	for name, data := range testCases {
+		t.Run(name, func(t *testing.T) {
+			var pkg Package
+			err := json.Unmarshal([]byte(data.JSON), &pkg)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !pkg.Equal(data.ExpectedPackage) {
+				t.Fatalf("expected:\n%#v got\n%#v", data.ExpectedPackage, pkg)
+			}
+			if data.ExpectedPackage.Location != nil {
+				if !pkg.Location.Equal(data.ExpectedPackage.Location) {
+					t.Fatalf("expected location:\n%#v got\n%#v", data.ExpectedPackage, pkg)
+				}
+			}
+		})
+	}
+}
+
+// TODO: Comment has inconsistent JSON field names starting with an upper case letter. Comment Location is
```

Comment:
[question]: Possibly a silly question, but is there a way for us to signal that the "legacy" capitalization is being deprecated, so that we can eventually change the field naming to the correct version?

Feel free to ignore if this is out-of-scope for this PR!

Reply:
I'm not sure how to do that in a good way, perhaps we could log to stderr but I'm not sure tbh.